sparkles/roll_optimize.py

```diff
@@ -185,7 +185,10 @@ def get_ids_list(roll_offsets):
                 FutureWarning
             )
         else:
-            roll_dev = Ska.Sun.allowed_rolldev(pitch) if max_roll_dev is None else max_roll_dev
+            # allowed_rolldev returns -1.0 for not-allowed pitch.
+            # Just clip to 0 in that case in the context of this function.
+            roll_dev = (ska_sun.allowed_rolldev(pitch).clip(0) if max_roll_dev is None
```

Comment:
The new behavior where allowed_rolldev returns -1.0 doesn't work here directly without modifying the return value, which I did by just clipping.  But I'm not sure if if this block needs rewrite, the whole function needs rewrite, or if the -1.0 return from allowed_rolldev should be reconsidered.

Reply:
This fix does not actually run for a pitch which is outside the allowed pitch range, failing in the same way that current flight code does in this situation. I took a different approach that seems to work and give expected results.

Reply:
Great! I like your logic better than my hack. Though my fix worked for me for the test_roll_outside_range test that was failing, so I'm a bit confused about it not actually working for you.  Hmm.  No matter.